modules/related-posts/jetpack-related-posts.php

```diff
@@ -1055,25 +1065,31 @@ protected function _get_related_post_ids( $post_id, $size, array $filters ) {
 			}
 		}
 
-		// Copy all valid cache values
-		$new_cache = array();
-		foreach ( $cache as $k => $v ) {
-			if ( is_array( $v ) && $v[ 'expires' ] > $now_ts ) {
-				$new_cache[ $k ] = $v;
+		$cache_ttl = 12 * HOUR_IN_SECONDS;
+
+		// Update cache
+		if ( wp_using_ext_object_cache() ) {
+			set_transient( $transient_name, $related_posts, $cache_ttl );
```

Comment:
Thanks @xyu 😄 I've set the cache TTL for an empty array to 1 minute. 1 minute is a short time, but gives some chance for sync and indexing to do their thing, and gives the Jetpack site relief from sending HTTP requests.

See https://github.com/Automattic/jetpack/pull/4391/commits/f7a47b85f575a5921739ae1e4ba800a4489fe2e4 and https://github.com/Automattic/jetpack/pull/4391/commits/a849c27b4973725602004a303a19aaed55bbe124